### PR TITLE
Add first input script

### PR DESCRIPTION
**Pull request by RemiLehe**
- [x] update test
- [x] add ParamParse logic in code
- [x] populate inputs script with a small number of elements

## `amrex::ParmParse`

Documentation: https://amrex-codes.github.io/amrex/docs_html/Basics.html#parmparse
Location in AMReX: https://github.com/AMReX-Codes/amrex/blob/development/Src/Base/AMReX_ParmParse.H

### CODE REVIEW

**Review by cemitch99 — COMMENTED** [review]
This looks good to me.  By the way, in IMPACT-Z, input files typically end in ".in", for example "test.in" or "input.in".

**Review by qianglbl — APPROVED** [review]

File: docs/source/usage/parameters.rst [hunk 1]
```diff
@@ -72,6 +72,31 @@ Domain Boundary Conditions
 
    TODO :-)
 
+Lattice elements
+----------------
```
Comment (RemiLehe) [inline docs/source/usage/parameters.rst:76]: @cemitch99 @qianglbl Feel free to suggest a better phrasing for the documentation below.
In particular, for the quad, maybe we should mention whether a positive `k` is focusing along `x` or along `y`.
For the bend, should we also specify along which direction it is bending the beam?

Note that, once the PR is merged, the text below will automatic appear on this page:
https://impactx.readthedocs.io/en/latest/usage/parameters.html

File: docs/source/usage/parameters.rst [hunk 2]
```diff
@@ -72,6 +72,31 @@ Domain Boundary Conditions
 
    TODO :-)
 
+Lattice elements
```
Comment (ax3l) [inline docs/source/usage/parameters.rst]: For consistency in titles:
```suggestion
Lattice Elements
```

File: src/ImpactX.cpp [hunk 3]
```diff
@@ -122,14 +122,36 @@ namespace impactx
...
-        // add elements
-        //   FODO cell
-        m_lattice.emplace_back(Quad(1.0, 4.0));
-        m_lattice.emplace_back(Drift(0.5));
-        m_lattice.emplace_back(Quad(1.0, 4.0));
-        m_lattice.emplace_back(Drift(0.5));
-        //   a bending magnet
-        m_lattice.emplace_back(Sbend(0.5, 2.0));
+        // Parse the lattice elements
+        amrex::ParmParse pp_lattice("lattice");
```
Comment (ax3l) [inline src/ImpactX.cpp:155]: Include missing: `#include <AMReX_ParmParse.H>`

File: src/ImpactX.cpp [hunk 4]
```diff
@@ -122,14 +122,36 @@ namespace impactx
...
+        std::vector<std::string> lattice_elements;
```
Comment (ax3l) [inline src/ImpactX.cpp:156]: Include missing: `#include <vector>`

File: src/ImpactX.cpp [hunk 5]
```diff
@@ -122,14 +122,36 @@ namespace impactx
...
+        pp_lattice.queryarr("elements", lattice_elements);
+
+        // Loop through lattice elements
+        for (std::string element_name : lattice_elements) {
```
Comment (ax3l) [inline src/ImpactX.cpp]: - Include missing: `#include <string>`
- `element_name` can be `const`

File: src/ImpactX.cpp [hunk 6]
```diff
@@ -122,14 +122,36 @@ namespace impactx
...
+            // Check the element type
+            amrex::ParmParse pp_element(element_name);
+            std::string element_type;
+            pp_element.get("type", element_type);
+            // Initialize the corresponding element according to its type
+            if (element_type == "quad") {
+                amrex::Real ds, k;
```
Comment (ax3l) [inline src/ImpactX.cpp:167]: Include missing: `#include <AMReX_REAL.H>`

**Review by ax3l — COMMENTED** [review]
Thanks a lot, looks great!

Added a few comments to improve code robustness (include directly used classes) and a comment on `const`-ness in a loop.

File: src/ImpactX.cpp [hunk 7]
```diff
@@ -4,14 +4,17 @@
...
+#include <vector>
+#include <string>
+
 #include "ImpactX.H"
 #include "particles/ImpactXParticleContainer.H"
 #include "particles/Push.H"
 
 #include <AMReX.H>
+#include <AMReX_REAL.H>
 #include <AMReX_ParmParse.H>
```
Comment (ax3l) [inline src/ImpactX.cpp:13]:
```suggestion
#include <AMReX_ParmParse.H>

#include <string>
#include <vector>

```

File: src/ImpactX.cpp [hunk 8]
```diff
@@ -4,14 +4,17 @@
  *
  * License: BSD-3-Clause-LBNL
  */
+#include <vector>
+#include <string>
+
```
Comment (ax3l) [inline src/ImpactX.cpp]: STL includes last, otherwise we cannot spot missing includes early
```suggestion
```

**Review by ax3l — APPROVED** [review]
Thanks a lot! :tada:

File: docs/source/usage/parameters.rst (hunk 1 above)
Comment (qianglbl) [reply docs/source/usage/parameters.rst:76]: It would probably be good to specify it in the document. (that should probably follow the same notation as other codes like MAD.
